src/main/java/gov/cms/mat/cql_elm_translation/service/HumanReadableService.java

```diff
@@ -98,57 +136,301 @@ HumanReadableMeasureInformationModel buildMeasureInfo(Measure measure) {
-                        .build())
-        .collect(Collectors.toList());
-  }
-
-  List<HumanReadablePopulationCriteriaModel> buildPopCriteria(Measure measure) {
+  List<HumanReadablePopulationCriteriaModel> buildPopCriteria(
+      Measure measure, Set<CQLDefinition> allDefinitions) {
     return measure.getGroups().stream()
         .map(
             group ->
                 HumanReadablePopulationCriteriaModel.builder()
                     .id(group.getId())
                     .name(group.getGroupDescription())
-                    .populations(buildPopulations(group))
+                    .populations(
+                        Stream.concat(
+                                buildPopulations(group, allDefinitions).stream(),
+                                buildStratification(group, allDefinitions).stream())
+                            .toList())
                     .build())
         .collect(Collectors.toList());
   }
 
-  private List<HumanReadablePopulationModel> buildPopulations(Group group) {
+  List<HumanReadablePopulationModel> buildPopulations(
+      Group group, Set<CQLDefinition> allDefinitions) {
     return group.getPopulations().stream()
         .map(
             population ->
                 HumanReadablePopulationModel.builder()
                     .name(population.getName().name())
                     .id(population.getId())
                     .display(population.getName().getDisplay())
-                    .logic(population.getDefinition())
+                    .logic(getCQLDefinitionLogic(population.getDefinition(), allDefinitions))
                     .expressionName(population.getDefinition())
+                    .inGroup(StringUtils.isBlank(population.getDefinition()) ? false : true)
+                    .build())
+        .collect(Collectors.toList());
+  }
+
+  String getCQLDefinitionLogic(String id, Set<CQLDefinition> allDefinitions) {
+    CQLDefinition cqlDefinition =
+        allDefinitions.stream()
+            .filter(definition -> id != null && id.equalsIgnoreCase(definition.getId()))
+            .findFirst()
+            .orElse(null);
+    return cqlDefinition != null
+        ? cqlDefinition.getLogic().substring(cqlDefinition.getLogic().indexOf('\n') + 1)
+        : "";
+  }
+
+  List<HumanReadablePopulationModel> buildStratification(
+      Group group, Set<CQLDefinition> allDefinitions) {
+    return group.getStratifications().stream()
+        .map(
+            stratification ->
+                HumanReadablePopulationModel.builder()
+                    .name("Stratification")
+                    .id(stratification.getId())
+                    .display("Stratification")
+                    .logic(getCQLDefinitionLogic(stratification.getCqlDefinition(), allDefinitions))
+                    .expressionName(stratification.getCqlDefinition())
+                    .inGroup(StringUtils.isBlank(stratification.getCqlDefinition()) ? false : true)
```

Comment:
Ternary can be removed.
```suggestion
                    .inGroup(!StringUtils.isBlank(stratification.getCqlDefinition()))
```

Reply:
Changed